### PR TITLE
Epic/nvid

### PR DESCRIPTION
**Pull request by mondain**

### CODE REVIEW

File: src/main/java/com/red5pro/server/plugin/simpleauth/datasource/impl/roundtrip/stream/security/PlaybackSecurity.java [hunk 1]
```diff
@@ -49,29 +54,42 @@
...
+				Iterator<Map.Entry<String, Object>> it = attrs.entrySet().iterator();
+				while (it.hasNext()) {
+					Map.Entry<String, Object> pair = it.next();
+					logger.debug("key : {} value : {}", pair.getKey(), pair.getValue());
+				}
+			}
+			return roundTripAuthValidator.onPlaybackAuthenticate(connection, scope, name);
 		}
+		// default playback result if a connection isn't present
+		return defaultResponse;
```
Comment (david179) [inline src/main/java/com/red5pro/server/plugin/simpleauth/datasource/impl/roundtrip/stream/security/PlaybackSecurity.java:88]: @mondain What kind of scenario could result in a null connection that allows to playback, or publish in PublishSecurity.java, by default?

File: src/main/resources/simple-auth-plugin.properties [hunk 2]
```diff
@@ -3,6 +3,12 @@
 # Default state of server wide security
 simpleauth.default.active=false
 
+# Default response for publish security
+simpleauth.default.publish.response=true
+
+# Default response for playback security
+simpleauth.default.playback.response=true
+
```
Comment (david179) [inline src/main/resources/simple-auth-plugin.properties:11]: @mondain The code doesn't seem to be reading these two parameters. Based on their names, PlaybackSecurity.java and PublishSecurity.java should read these to set `defaultResponse`

File: src/main/java/com/red5pro/server/plugin/simpleauth/datasource/impl/roundtrip/stream/security/PlaybackSecurity.java (hunk 1 above)
Comment (mondain) [reply src/main/java/com/red5pro/server/plugin/simpleauth/datasource/impl/roundtrip/stream/security/PlaybackSecurity.java:88]: A null connection can happen when connection local is improperly handled as well as when ConnectorShell is in-use (not 100% on that last one)

File: src/main/resources/simple-auth-plugin.properties (hunk 2 above)
Comment (mondain) [reply src/main/resources/simple-auth-plugin.properties:11]: @david179 that seems logical to me

**Comment by mondain** [comment]
PR #16 supersedes this one.

**Pull request closed without merging**